libpod/container.go

```diff
@@ -1,76 +1,312 @@
 package libpod
 
 import (
+	"encoding/json"
+	"io/ioutil"
+	"path/filepath"
 	"sync"
+	"time"
 
 	"github.com/containers/storage"
 	"github.com/docker/docker/pkg/stringid"
+	crioAnnotations "github.com/kubernetes-incubator/cri-o/pkg/annotations"
 	spec "github.com/opencontainers/runtime-spec/specs-go"
 	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
 	"github.com/ulule/deepcopier"
 )
 
+// ContainerState represents the current state of a container
+type ContainerState int
+
+const (
+	// ContainerStateUnknown indicates that the container is in an error
+	// state where information about it cannot be retrieved
+	ContainerStateUnknown ContainerState = iota
+	// ContainerStateConfigured indicates that the container has had its
+	// storage configured but it has not been created in the OCI runtime
+	ContainerStateConfigured ContainerState = iota
+	// ContainerStateCreated indicates the container has been created in
+	// the OCI runtime but not started
+	ContainerStateCreated ContainerState = iota
+	// ContainerStateRunning indicates the container is currently executing
+	ContainerStateRunning ContainerState = iota
+	// ContainerStateStopped indicates that the container was running but has
+	// exited
+	ContainerStateStopped ContainerState = iota
+	// ContainerStatePaused indicates that the container has been paused
+	ContainerStatePaused ContainerState = iota
+)
```

Comment:
Just curios.  Do you need "ContainerState = iota" in any of the lines except line 25?  I'm just not sure if it resets to 0 each time or not?

Reply:
I believe it's required since the lines aren't actually sequential. If I didn't have comments between each constant I wouldn't need it. It does not reset from 0 within a single block, so numbering is properly incrementing.

Reply:
FWIW, I just did a quick prototype and it works as is or with them removed, even with the comments.  Leaving as is is fine by me.